dtable.c

```diff
@@ -824,26 +824,31 @@ PRIVATE void objc_send_initialize(id object)
 		objc_send_initialize((id)class->super_class);
 	}
 
+	// Lock the runtime while we're creating dtables and before we acquire any
+	// other locks.  This prevents a lock-order reversal when
+	// dtable_for_class is called from something holding the runtime lock while
+	// we're still holding the initialize lock.  We should ensure that we never
+	// acquire the runtime lock after acquiring the initialize lock.
+	LOCK_RUNTIME();
```

Comment:
I'm not sure. We want to drop the lock early before exiting the scope ([l. 876](https://github.com/ngrewe/libobjc2/blob/6b285936bed090d06f3bfc66d0886c12dc088753/dtable.c#L876)) to avoid that blocking unrelated runtime operations while +initialize runs for the class.